app/src/main/java/com/jHerscu/clearskies/domain/repoInterface/PreferencesRepo.kt

```diff
@@ -1,11 +1,96 @@
 package com.jHerscu.clearskies.domain.repoInterface
 
+import androidx.annotation.StringRes
 import androidx.datastore.preferences.core.Preferences
+import com.jHerscu.clearskies.R
+import com.jHerscu.clearskies.ui.prefs.components.RadioComponentData
 import kotlinx.coroutines.flow.Flow
 
+enum class TempUnitPref(override val labelRes: Int) : RadioComponentData {
+    FAHRENHEIT(R.string.degrees_fahrenheit),
+    CELSIUS(R.string.degrees_celcius)
+}
+
+enum class LockedThemePref(override val labelRes: Int) : RadioComponentData {
+    LOCK_DARK(R.string.dark_mode),
+    LOCK_LIGHT(R.string.light_mode),
+    UNLOCKED(R.string.unlocked)
+}
+
+enum class HomeDisplayIntervalPref(override val labelRes: Int) : RadioComponentData {
+    HOURLY(R.string.hourly),
+    DAILY(R.string.daily)
+}
+
+enum class PreferenceGroup(@StringRes val titleRes: Int) {
+    THEME(R.string.theme_pref_group_title),
+    GENERAL(R.string.general_pref_group_title),
+    HOME(R.string.home_screen)
+    // TODO(jherscu): NOTIFICATIONS(R.string.notifications_pref_group_title)
+    ;
+
+    companion object {
+        val defaultSortOrder = listOf(GENERAL, THEME, HOME)
+    }
+}
+
+data class TranslatedPrefGroup(
+    val translatedTitle: String,
+    val prefGroup: PreferenceGroup
+)
+
+enum class PreferenceOrderComparator(
+    val titleRes: Int,
+    val sortingFun: (List<TranslatedPrefGroup>) -> List<PreferenceGroup>
+) { // TODO(jherscu): add sorting usage
+    DEFAULT(
+        R.string.default_sort,
+        { _ -> PreferenceGroup.defaultSortOrder }
+    ),
+    MOST_CLICKED(
+        R.string.most_clicked_sort,
+        { _ ->
+            emptyList() // TODO(jherscu): instument interactions by card to prefs to form sort order
+        }
+    ),
+    ALPHABETICAL( // Use translated strings as alphabetical could change based on language
+        R.string.alphabetical_sort,
+        { translatedPrefGroup ->
+            translatedPrefGroup
```

Comment:
look into moving data classes/enums to model and rm RadioComponentData interface

Reply:
done